### PR TITLE
Fix memory leak issue

### PR DESCRIPTION
**Pull request by alr2413**
Recently, @mitchross and I found there are situations that all variables & objects employed in vlc_player_controller are not disposed completely.
This could cause memory leak issue. This PR is a simple fix for the problem.

### CODE REVIEW

**Comment by mitchross** [comment]
I tested this in a side project. This resolves issue on Android. I tested on iOS and confirmed no issues. Pending no implementation PR feedback, this is safe to merge. @solid-software

**Comment by alr2413** [comment]
I should mention that we also have another solution to fix this issue. let's first talk about why this issue happens.

in android, we implemented the platformView class for managing the vlc player actions. when you dispose the widget on dart, the dispose method of platformView class in android is being called automatically, so all object reference in android are getting destroyed while in dart we are not aware of that. As a result, when we call controller dispose on dart, it throws exceptions that the communication platform channel does no longer exist or VLC object is null, thus some objects are not disposed properly in dart which cause the memory leak issue.

To overcome this problem, we can use the submitted PR or we can do the following:

in dart -> revert everything on dispose method to the previous state.
in android -> make the body of override dispose method empty and move all of the lines to another method and call that method to dispose all references.

By using this way, we handle the disposition process from dart side and we no longer need the isDisposed variable on android code.

Please, share you thoughts with me.
Thanks

File: flutter_vlc_player/example/lib/single_tab.dart [hunk 1]
```diff
@@ -221,9 +222,10 @@ class _SingleTabState extends State<SingleTab> {
   }
 
   @override
-  void dispose() {
-    _controller.stopRendererScanning();
-    _controller.removeListener(() {});
+  void dispose() async {
     super.dispose();
+    _controller.removeListener(() {});
```
Comment (solid-yuriiprykhodko) [inline flutter_vlc_player/example/lib/single_tab.dart:227]: This line does exactly nothing.
We can remove a listener after we've added one, and using exactly the same reference that we used when adding it.

We should also remove a similar line in `README.md` and in `multiple_tab.dart`

File: flutter_vlc_player/example/lib/multiple_tab.dart [hunk 2]
```diff
@@ -64,11 +64,12 @@ class _MultipleTabState extends State<MultipleTab> {
   }
 
   @override
-  void dispose() {
+  void dispose() async {
+    super.dispose();
     for (var i = 0; i < controllers.length; i++) {
```
Comment (solid-yuriiprykhodko) [inline flutter_vlc_player/example/lib/multiple_tab.dart:69]: Let's use a `for-in` loop here - we don't need to use indices here.
```suggestion
        for (final controller in controllers) {
```

File: flutter_vlc_player/example/lib/single_tab.dart [hunk 3]
```diff
@@ -181,15 +182,15 @@ class _SingleTabState extends State<SingleTab> {
                         hwAcc: HwAcc.FULL);
                     break;
                   case VideoType.file:
-                    Scaffold.of(context).showSnackBar(
+                    ScaffoldMessenger.of(context).showSnackBar(
```
Comment (solid-yuriiprykhodko) [inline flutter_vlc_player/example/lib/single_tab.dart:185]: The `ScaffoldMessenger` breaking change hasn't been introduced into Flutter stable channel yet. Perhaps we should hold off this change until then?

**Review by solid-yuriiprykhodko — CHANGES_REQUESTED** [review]
Good catch! Memory usage indeed looks slightly smaller (~ -5MB by my measurements).
A few minor changes requested.

**Comment by mitchross** [comment]
@solid-yuriiprykhodko I tested iOS and didnt see any memory issues either. If you notice anything wrong with iOS please let me know.

**Comment by alr2413** [comment]
@solid-yuriiprykhodko, so you think this PR is ok or i should do the following instead:

```
To overcome this problem, we can use the submitted PR or we can do the following:

in dart -> revert everything on dispose method to the previous state.
in android -> make the body of override dispose method empty and move all of the lines to another method and call that method to dispose all references.
```

I'm waiting for your response?

**Review by solid-yuriiprykhodko — APPROVED** [review]
LGTM!

**Pull request merged**